Examples/test-suite/fortran/Makefile.in

```diff
@@ -0,0 +1,126 @@
+#######################################################################
+# Makefile for Fortran test-suite
+#######################################################################
+
+LANGUAGE	= fortran
+FC      	= @FC@
+SCRIPTSUFFIX= _runme.f90
+
+FORTRAN_SO = @FORTRAN_SO@
+FCFLAGS = @FCFLAGS@
+CCSHARED=	@CCSHARED@
+
+srcdir         = @srcdir@
+top_srcdir     = @top_srcdir@
+top_builddir   = @top_builddir@
+
+CPP_TEST_CASES = \
+	fortran_bindc
+
+CPP11_TEST_CASES = \
+	fortran_shared_ptr
+
+C_TEST_CASES = \
+	fortran_array_typemap \
+	fortran_bindc_c \
+	fortran_bindc_all \
+	fortran_enums \
+
+# C++11 tests: wrapper error
+FAILING_CPP_TESTS += \
+  cpp11_lambda_functions \
+  cpp11_rvalue_reference2 \
+  cpp11_rvalue_reference3 \
+  cpp11_template_explicit
+
+# C++11 tests: link error
+FAILING_CPP_TESTS += \
+  cpp11_default_delete
+
+# C++11 tests: fortran error
+FAILING_CPP_TESTS += \
+  cpp11_strongly_typed_enumerations
+
+# Private 'operator new' causes build error since it's not detected by the type
+# traits
+ifneq (,$(HAVE_CXX11_COMPILER))
```

Comment:
The `AssignmentTraits` and `SWIG_assign_impl` and all those associated terrible gymnastics are the only solution I can come up with to enabling the assignment operator in Fortran (see `Fortran.html#Fortran_memory_management`). The SWIG-created assignment operator has to be able to copy-construct, assign, and/or delete the target, depending on the states of the left and right hand sides of the equation. Constructing traits based on the capabilities of the LHS and RHS classes seemed like the only feasible approach. It also allows move-construction in C++11, which is important since Fortran is targeted at high-performance scientific computation where the data structures (and the penalty of copying them) is enormous.

Reply:
I have simplified the logic in this class to eliminate C++11 specialization and rely more on SWIG introspection. This exception has been removed from the makefile.